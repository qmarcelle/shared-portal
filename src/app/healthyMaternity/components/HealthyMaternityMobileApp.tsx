import { Card } from '@/components/foundation/Card';
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { ListOrder } from '@/components/foundation/ListOrder';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import Image from 'next/image';
import Link from 'next/link';
import careTNDevice from '../../../../public/assets/caretn_device.png';

const HealthyMaternityMobileApp = () => {
  return (
    <>
      <Card className="large-section overlap-bottom">
        <section className="md:flex md:flex-row">
          <Image src={careTNDevice} alt="" />
          <Spacer size={16} />
          <Column>
            <Header
              text="Healthy Maternity & CareTN Mobile App"
              type="title-2"
            />
            <Spacer size={16} />
            <TextBox text="Once you've enrolled, download the CareTN app. You can use the app to message a maternity nurse and access program resources." />
            <ListOrder
              title="In the app you can:"
              itemData={[
                'Learn about eating healthy for you and your baby',
                'Track doctor visits and set reminders',
                'Get information on immunizations and when to schedule them',
                'Connect with maternity nurse',
              ]}
            />
            <Spacer size={32} />
            <section className="justify-start flex flex-row">
              <Link href="https://apps.apple.com/us/app/amplifyhealth/id6444489335">
                <Image src={appleStoreLogo} alt="" />
              </Link>
              <Link
                className="pl-[10px]"
                href="https://play.google.com/store/apps/details?id=com.bcbst.amplify&hl=en_US"
              >
                <Image src={playStoreLogo} alt="" />
              </Link>
            </section>
          </Column>
        </section>
      </Card>
    </>
  );
};

export default HealthyMaternityMobileApp;
