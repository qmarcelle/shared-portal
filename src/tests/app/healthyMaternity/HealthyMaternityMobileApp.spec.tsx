import HealthyMaternityMobileApp from '@/app/healthyMaternity/components/HealthyMaternityMobileApp';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(<HealthyMaternityMobileApp />);
};

describe('HealthyMaternityMobileApp Section', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    expect(
      screen.getByText('Healthy Maternity & CareTN Mobile App'),
    ).toBeVisible();
    expect(
      screen.getByText(
        // eslint-disable-next-line quotes
        "Once you've enrolled, download the CareTN app. You can use the app to message a maternity nurse and access program resources.",
      ),
    ).toBeVisible();
    expect(
      screen.getByText('Learn about eating healthy for you and your baby'),
    ).toBeVisible();
    expect(
      screen.getByText('Track doctor visits and set reminders'),
    ).toBeVisible();
    expect(
      screen.getByText(
        'Get information on immunizations and when to schedule them',
      ),
    ).toBeVisible();
    expect(screen.getByText('Connect with maternity nurse')).toBeVisible();
    expect(component).toMatchSnapshot();
  });
});
