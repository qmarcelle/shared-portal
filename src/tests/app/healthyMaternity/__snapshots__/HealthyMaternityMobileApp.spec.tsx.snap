// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthyMaternityMobileApp Section should render the UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-main large-section overlap-bottom"
      >
        <section
          class="md:flex md:flex-row"
        >
          <img
            alt="careTN"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="flex flex-col"
          >
            <h2
              class="title-2"
            >
              Healthy Maternity & CareTN Mobile App
            </h2>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
            <p
              class="body-1"
              style="display: block;"
            >
              Once you've enrolled, download the CareTN app. You can use the app to message a maternity nurse and access program resources.
            </p>
            <div>
              <p
                class="body-1 body-1 m-4"
                style="display: block;"
              >
                In the app you can:
              </p>
              <ul
                class="list-disc m-8 mb-0 marker:text-[#5DC1FD]"
              >
                <div>
                  <li>
                    Learn about eating healthy for you and your baby
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Track doctor visits and set reminders
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Get information on immunizations and when to schedule them
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Connect with maternity nurse
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
              </ul>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
            />
            <section
              class="justify-start flex flex-row"
            >
              <a
                href="https://apps.apple.com/us/app/amplifyhealth/id6444489335"
              >
                <img
                  alt="Download Amplify App from AppStore"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
              <a
                class="pl-[10px]"
                href="https://play.google.com/store/apps/details?id=com.bcbst.amplify&hl=en_US"
              >
                <img
                  alt="Download Amplify App from PlayStore"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </section>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-main large-section overlap-bottom"
    >
      <section
        class="md:flex md:flex-row"
      >
        <img
          alt="careTN"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col"
        >
          <h2
            class="title-2"
          >
            Healthy Maternity & CareTN Mobile App
          </h2>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <p
            class="body-1"
            style="display: block;"
          >
            Once you've enrolled, download the CareTN app. You can use the app to message a maternity nurse and access program resources.
          </p>
          <div>
            <p
              class="body-1 body-1 m-4"
              style="display: block;"
            >
              In the app you can:
            </p>
            <ul
              class="list-disc m-8 mb-0 marker:text-[#5DC1FD]"
            >
              <div>
                <li>
                  Learn about eating healthy for you and your baby
                </li>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
              </div>
              <div>
                <li>
                  Track doctor visits and set reminders
                </li>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
              </div>
              <div>
                <li>
                  Get information on immunizations and when to schedule them
                </li>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
              </div>
              <div>
                <li>
                  Connect with maternity nurse
                </li>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
              </div>
            </ul>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <section
            class="justify-start flex flex-row"
          >
            <a
              href="https://apps.apple.com/us/app/amplifyhealth/id6444489335"
            >
              <img
                alt="Download Amplify App from AppStore"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </a>
            <a
              class="pl-[10px]"
              href="https://play.google.com/store/apps/details?id=com.bcbst.amplify&hl=en_US"
            >
              <img
                alt="Download Amplify App from PlayStore"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </a>
          </section>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
